api/file.go

```diff
@@ -195,47 +201,56 @@ func doUploadFile(teamId string, channelId string, userId string, rawFilename st
 func handleImages(previewPathList []string, thumbnailPathList []string, fileData [][]byte) {
 	for i, data := range fileData {
 		go func(i int, data []byte) {
-			// Decode image bytes into Image object
-			img, imgType, err := image.Decode(bytes.NewReader(fileData[i]))
-			if err != nil {
-				l4g.Error(utils.T("api.file.handle_images_forget.decode.error"), err)
-				return
+			preparedImage := prepareImage(fileData[i])
+			if preparedImage != nil {
+				go generateThumbnailImage(preparedImage.Img, thumbnailPathList[i], preparedImage.Width, preparedImage.Height)
+				go generatePreviewImage(preparedImage.Img, previewPathList[i], preparedImage.Width)
 			}
+		}(i, data)
+	}
+}
+
+func prepareImage(fileData []byte) *PreparedImage {
```

Comment:
Maybe just make this return multiple values instead of making a struct that's only used by one function?
```
func prepareImage(fileData []byte) (*image.Image, int, int) {
```